staging/src/k8s.io/apiserver/pkg/registry/generic/rest/streamer_test.go

```diff
@@ -146,3 +148,23 @@ func TestInputStreamInternalServerErrorTransport(t *testing.T) {
 		t.Errorf("StreamInternalServerError does not match. Got: %s. Expected: %s.", err, expectedError)
 	}
 }
+
+func TestInputStreamRedirects(t *testing.T) {
+	redirected := false
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if redirected {
+			w.Write([]byte("ok"))
+			redirected = true
+		} else {
+			http.Redirect(w, req, "/result", http.StatusFound)
+		}
+	}))
+	loc, err := url.Parse(s.URL)
+	require.NoError(t, err, "Error parsing server URL")
+
+	streamer := &LocationStreamer{
+		Location: loc,
+	}
+	_, _, _, err = streamer.InputStream("", "")
+	assert.Equal(t, http.ErrUseLastResponse, err, "Expected error from redirect")
```

Comment:
If CheckRedirect returns ErrUseLastResponse, then the most recent response is returned with its body unclosed, along with a nil error: https://github.com/golang/go/blob/214f7ec554888042a7a54fdca9cba19aee8ebaf1/src/net/http/client.go#L583-L588

Reply:
Oops, thanks. On closer inspection, forwarding the redirect doesn't make sense here since only the request body is returned. I reverted the change back to outright rejecting redirects.